File: daily/2025/2176_Count_Equal_and_Divisible_Pairs_in_an_Array.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int countPairs(vector<int>& nums, int k) {
        int n = nums.size();
        int ans = 0;
        for (int i = 0; i < n; i++){
            for (int j = i + 1; j < n; j++){
                if (nums[i] == nums[j] && i * j % k == 0) ans++;
            }
        }
        return ans;
    }
};

int main(){
    vector<int> nums = {3,1,2,2,2,1,3};
    int k = 2;
    // vector<int> nums = {1,2,3,4};
    // int k = 1;
    Solution solution;
    cout << solution.countPairs(nums, k) << endl;
    return 0;
}

File: daily_python/2176_Count_Equal_and_Divisible_Pairs_in_an_Array.py
import sys
import os
from typing import List

class Solution:
    def countPairs(self, nums: List[int], k: int) -> int:
        n = len(nums)
        count = 0
        for i in range(n):
            for j in range(i + 1, n):
                if nums[i] == nums[j] and (i * j) % k == 0:
                    count += 1
        return count
    
def readData(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            nums = list(map(int, lines[0].split('=')[1].strip()[1:-2].split(',')))
            k = int(lines[1].split('=')[1].strip()[:-1])
            dataset.append((nums, k))
    return dataset

if __name__ == '__main__':
    if (len(sys.argv) == 1):
        path = os.path.basename(__file__)
        filename = os.path.splitext(path)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readData(filename)
    results = []
    solution = Solution()
    for nums, k in dataset:
        results.append(solution.countPairs(nums, k))
    for index, result in enumerate(results):
        print(f'dataset {index+1} : {result}')
